### ISSUE TITLE
Windows support broken by periodic block generation

### ISSUE THREAD

**Issue by FabijanC** [0xSpaceShard/starknet-devnet-rs#561]
So far it seems Windows users have been able to use Devnet via `cargo install`. But after https://github.com/0xSpaceShard/starknet-devnet-rs/commit/24ed52d435e995f80947dd622450d256709c9577, this has not been the case.

Find more info in [this Discord thread](https://discord.com/channels/793094838509764618/1263752663892492380).

Please, before we dismiss this with "Windows support not planned", let's try considering an alternative to using `tokio::signal::unix` in `main.rs`, which is the culprit.

**Comment by mikiw** [comment]
I think that configuration check can be added like in `send_ctrl_c_signal` with `https://docs.rs/tokio/latest/tokio/signal/windows/index.html`

```
async fn send_ctrl_c_signal(process: &Child) {
    #[cfg(windows)]
    {
        // To send SIGINT signal on windows, windows-kill is needed
        let mut kill = Command::new("windows-kill")
            .args(["-SIGINT", process.id().to_string().as_str()])
            .spawn()
            .unwrap();
        kill.wait().unwrap();
    }

    #[cfg(unix)]
    {
        let mut kill = Command::new("kill")
            .args(["-s", "SIGINT", process.id().to_string().as_str()])
            .spawn()
            .unwrap();
        kill.wait().unwrap();
    }
}
```

**Comment by FabijanC** [comment]
Closed via #565.
